### PR TITLE
Refactor em-app to support nitro enclaves and others

### PR DESCRIPTION
**Pull request by AdrianCX**
Code is mostly moved around.
- CSR code is now in a separate csr.rs file
- errors in a separate error.rs
- separate platform folder where each platform implements things differently. (new file - nitro.rs)

Note:
- platform is selected via feature right now to allow multiple targets, can switch to target specific.

### CODE REVIEW

**Review by ssavvides — APPROVED** [review]

File: em-app/src/platform/mod.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#[cfg(feature = "sgx")]
```
Comment (ssavvides) [inline em-app/src/platform/mod.rs:1]: I am not sure how thorough we should be with copyright, but in any case, copyright is missing from a few files.

File: em-app/Cargo.toml [hunk 2]
```diff
@@ -12,25 +12,39 @@ categories = [ "api-bindings" ]
 keywords = [ "sgx" ]
 
 [dependencies]
-hyper = { version = "0.10", default-features = false }
-mbedtls = { version = "0.8.1", features = [ "rdrand" ] }
 b64-ct = "0.1.0"
```
[the remark this answers is not in the document]
Comment (AdrianCX) [reply em-app/Cargo.toml:15]: b64-ct - for constant time was required in a previous PR, won't change to rustc-serialize

File: em-app/Cargo.toml [hunk 3]
```diff
@@ -12,25 +12,39 @@ categories = [ "api-bindings" ]
...
-rustc-serialize = { git = "https://github.com/jethrogb/rustc-serialize.git", branch = "portability" }
-
 em-client = { git = "https://github.com/fortanix/em-client-rust", branch = "acruceru/api-update", default-features = false, features = ["client"] }
 em-node-agent-client = "1.0.0"
-sgx_pkix = "0.1.0"
-pkix = "0.1.1"
+hyper = { version = "0.10", default-features = false }
+lazy_static = "1"
+mbedtls = { version = "0.8.1", features = [ "rdrand", "std" ], default-features = false }
+pkix = { git = "https://github.com/fortanix/pkix", branch="acruceru/ecdsa_csr" }
```
[the remark this answers is not in the document]
Comment (AdrianCX) [reply em-app/Cargo.toml:21]: I'll raise a PR there soon.

[the remark this answers is not in the document]
Comment (AdrianCX) [reply em-app/Cargo.toml:21]: https://github.com/fortanix/pkix/pull/8/files

[the remark this answers is not in the document]
Comment (ssavvides) [reply em-app/Cargo.toml:21]: done

**Comment by AdrianCX** [comment]
raised separate PR as follow up, will close this: https://github.com/fortanix/rust-sgx/pull/379

**Pull request closed without merging**